### ISSUE TITLE
If you sum a List[BigDecimal] the compiler can not decide which implicit to use

### ISSUE THREAD

**Issue by DavidGoodenough** [lampepfl/dotty#12737]
scala 3.0.0

```scala
val list:List[BigDecimal] = List(BigDecimal(2),BigDecimal(3))
println(list.sum.toString)
```

This generated a compiler error:-
"ambiguous implicit arguments: both object DoubleIsFractional in object Numeric and object BigDecimalIsFractional in object Numeric match type Numeric[B] of parameter num of method sum in trait IterableOnceOps"

```scala
println(list.sum(BigDecimalIsFractional).toString) 
```

works.

The orginal code  compiled under Scala-2, but under Scala-3 it fails with this error

**Comment by som-snytt** [comment]
I am unable to reproduce the error.
```
➜  snips scala
scala> val list:List[BigDecimal] = List(BigDecimal(2),BigDecimal(3))
val list: List[BigDecimal] = List(2, 3)

scala> list.sum.toString
val res0: String = 5

scala>
➜  snips ~/projects/dotty/bin/scala
scala> val list:List[BigDecimal] = List(BigDecimal(2),BigDecimal(3))
val list: List[BigDecimal] = List(2, 3)

scala> list.sum.toString
val res0: String = 5
```

**Comment by prolativ** [comment]
I cannot reproduce this either. @DavidGoodenough did you get this error while running the code in some specific way (e.g. something evaluated in the same REPL session before, using some imports not included in the snippet, some compiler flags added)?

**Comment by DavidGoodenough** [comment]
I was using vscodium and metals, but it may be a presentation compiler problem as I only got when I have lots or errors and the BSP times out

**Comment by prolativ** [comment]
So the presentation compiler complains but the project does actually compile OK?

**Comment by DavidGoodenough** [comment]
On Tuesday, 8 June 2021 14:15:09 BST Michał Pałka wrote:
> So the presentation compiler complains but the project does 
actually compile
> OK?
No.  When metals can not communicate with the BSP because of 
the timeout the presentation compiler fails.  The project does not 
compile.  When I resolve enough of the errors (I am migrating code 
from Scala-2 to Scala-3 so there are a number) down to the point 
where the BSP no longer times out then the real compiler takes 
over and this error goes away.

**Comment by prolativ** [comment]
Ok, so the core issue is actually completely different to the one described at the top.
Have you looked at the metals logs? Unless they contain something indicating that this is definitely a problem with scala 3 compiler itself, this issue should be closed and you should open a new one in metals repo

**Comment by DavidGoodenough** [comment]
On Tuesday, 8 June 2021 14:26:35 BST Michał Pałka wrote:
> Ok, so the core issue is actually completely different to the one described
> at the top. Have you looked at the metals logs? Unless they contain
> something indicating that this is definitely a problem with scala 3
> compiler itself, this issue should be closed and you should open a new one
> in metals repo
Yes, when I first hit it I did not realise that I was using just the presentation compiler.  
It was only after it was noticed that others could not reproduce it that I reversed the 
explicit implicit and found that it now worked.  I am assuming that this was because 
of the transition to proper BSP, but I am not sure.

If I hit is again I will take a deeper look. 

So yes for now this can be closed.
